libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -406,6 +413,161 @@ impl Greenlight {
+                short_channel_id: hop.channel.clone(),
+                channel_delay: first_channel.delay as u64,
+            });
+        }
+        Ok(PaymentPath { edges: hops })
+    }
+
+    async fn max_sendable_amount_from_peer(
+        &self,
+        via_peer_id: Vec<u8>,
+        via_peer_channels: Vec<ListpeersPeersChannels>,
+        payee_node_id: Option<Vec<u8>>,
+        max_hops: u32,
+        last_hop_hints: Option<&RouteHintHop>,
+    ) -> NodeResult<Vec<MaxChannelAmount>> {
+        let mut client = self.get_node_client().await?;
+
+        // Consider the hints as part of the route. If there is a routing hint we will
+        // attempt to calculate the path until the last hop in the hint and then add
+        // the last hop to the path.
+        let (last_node, max_hops) = match last_hop_hints {
+            Some(hop) => (hex::decode(&hop.src_node_id)?, max_hops - 1),
+            None => match payee_node_id.clone() {
+                Some(node_id) => (node_id, max_hops),
+                None => {
+                    return Err(NodeError::RouteNotFound(anyhow!(
+                        "No payee node id or last hop hints provided, cannot calculate max amount"
+                    )));
+                }
+            },
+        };
+
+        // fetch a route from greenlight
+        let route_result = client
+            .get_route(GetrouteRequest {
+                id: last_node.clone(),
+                amount_msat: Some(Amount { msat: 0 }),
+                riskfactor: 0,
+                cltv: None,
+                fromid: Some(via_peer_id.clone()),
+                fuzzpercent: Some(0),
+                exclude: vec![],
+                // we deduct the first hop that we calculate manually
+                maxhops: Some(max_hops - 1),
+            })
+            .await;
+
+        // In case we have no route better to return no amounts for this peer's channels.
+        if let Err(e) = route_result {
+            error!(
+                "Failed to get route for peer {}: {}",
+                hex::encode(via_peer_id.clone()),
+                e
+            );
+            return Ok(vec![]);
+        }
+
+        let route_response = route_result?.into_inner();
+        info!(
+            "max_sendable_amount: route response = {:?}",
+            route_response.route
+        );
+
+        // We fetch the opened channels so can calculate max amount to send for each channel
+        let mut opened_channels: Vec<cln::ListpeersPeersChannels> = via_peer_channels
+            .iter()
+            .cloned()
+            .filter(|c| c.state() == ChanneldNormal)
+            .collect();
+
+        // opened_channels.into_iter().filter(|c: &cln::ListpeersPeersChannels|c)
+        opened_channels.sort_by(|c1, c2| {
+            c2.spendable_msat
+                .clone()
+                .unwrap_or_default()
+                .msat
+                .cmp(&c1.spendable_msat.clone().unwrap_or_default().msat)
+        });
+
```

Comment:
This can be removed. (re https://github.com/breez/breez-sdk/pull/466#discussion_r1383148636)
```suggestion
```

Reply:
See my comment there, it is still needed.